### DISCUSSION TITLE
Strange combination of handshake error with undefined log error

### DISCUSSION THREAD

**Discussion by haydenyoung** [libp2p/js-libp2p#2265] [Q&A] [answered]
I have two vue3 projects dialing the same relay. I have both pretty much with identical configuration. One connects to the relay without issue and returns the discovered address. The other errors with:

```
Uncaught (in promise) CodeError: TypeError: Error occurred during XX handshake: options.log is undefined
```

The relay shows the following errors when the "bad" peer connects:

```
  libp2p:upgrader:error Failed to upgrade inbound connection CodeError: Error: stream ended before 1 bytes became available

```

and 

```
  libp2p:websockets:listener:error inbound connection failed to upgrade CodeError: Error: stream ended before 1 bytes became available

```

It looks like the error is occurring during the communication encryption process as the line before the errors indicates a noise() upgrade:

```
libp2p:upgrader handling inbound crypto protocol selection [ '/noise' ] +0ms
```

The only way to fix this seems to be to copy the @libp2p and libp2p directories from node_modules from the "good" project to the "bad" project which would suggest there is an issue with one of the libp2p libs, although I haven't been able to isolate which one (I'm guessing either noise or one of its dependencies). I have also diff-ed both the good and bad projects' node_modules/libp2p and node_modules/@libp2p and there are version discrepancies between both. I'm wondering if a minor update has caused the above errors to appear.

For reference: 0.46.18 works. Any version above this causes the above errors, both libp2p 0.46.x where x > 18 and 1.0.0.

A simple example demonstrating the issue can be provided if necessary.

UPDATE: I made a copy of the "good" project and deleted the node_modules and package-lock.json. After running `npm i` again to install libp2p and re-running the "good" project, it is now displaying the same error as above so it seems something has broken in one of the package updates.

**Comment by achingbrain** [comment, marked as answer]
There was a regression in `@libp2p/multistream-select@4.0.7` that caused the `options.log` error - it's since been fixed.

Please can you delete your lockfiles, reinstall your deps, and ensure you have `@libp2p/multistream-select@4.0.8` or later then give it another go?

**Comment by haydenyoung** [comment]
Thanks. That fixed the problem.
